### PR TITLE
Add `sphinxcontrib.jquery` extension to `conf.py`

### PR DESCRIPTION
**Pull request by Alnusjaponica**
## Motivation
In the latest optuna document, the search result page shows "Searching..." but that it never ends. In addition, the menu tab in the lower left part doesn't pull up. 
This PR resolves these problems.

## Description of the changes

As written in https://github.com/readthedocs/sphinx_rtd_theme/issues/1451, the combination of the latest versions of  `sphinx` and `sphinx_rtd_theme` causes error in search result page. This PR adds `sphinxcontrib.jquery` extension in `conf.py` to make `jQuery` loaded.

### CODE REVIEW

**Comment by Alnusjaponica** [comment]
This workaround might be possible to remove after https://github.com/readthedocs/sphinx_rtd_theme/issues/1452 is resolved.

File: docs/source/conf.py [hunk 1]
```diff
@@ -55,6 +55,7 @@
     "sphinx.ext.napoleon",
     "sphinx.ext.viewcode",
     "sphinx.ext.githubpages",
+    "sphinxcontrib.jquery",
```
Comment (toshihikoyanase) [inline docs/source/conf.py:58]: As you mentioned in https://github.com/optuna/optuna/pull/4615#issuecomment-1510024686, this fix is a workaround for https://github.com/readthedocs/sphinx_rtd_theme/issues/1452. How about adding a TODO comment to remove it after resolving the issue?

**Review by toshihikoyanase — COMMENTED** [review]
I confirmed that the rendered reference worked as expected, including the search window and the version selection. I have a small suggestion.

**Review by not522 — APPROVED** [review]
LGTM except for @toshihikoyanase's comment.

File: docs/source/conf.py (hunk 1 above)
Comment (Alnusjaponica) [reply docs/source/conf.py:59]: Thank you for your indication. I added a TODO comment.

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your update. LGTM!

**Pull request merged**